pkg/controller/routes.go

```diff
@@ -35,6 +35,7 @@ func routes(router *router.Router, registryTransport http.RoundTripper) {
 	router.HandleFunc(&v1.AppInstance{}, appdefinition.PullAppImage(registryTransport))
 	router.HandleFunc(&v1.AppInstance{}, appdefinition.ParseAppImage)
 	router.HandleFunc(&v1.AppInstance{}, tls.ProvisionCerts) // Provision TLS certificates for port bindings with user-defined (valid) domains
+	router.HandleFunc(&v1.AppInstance{}, appdefinition.AddNamespace)
```

Comment:
Thanks for catching this! 

It is already addressed by tests here 

https://github.com/acorn-io/acorn/blob/b00dc8caad372f184116414fcf1c1fbd3941bdb3/pkg/controller/appdefinition/testdata/deployspec/no-user-labels-namespace/input.yaml#L18 

https://github.com/acorn-io/acorn/blob/b00dc8caad372f184116414fcf1c1fbd3941bdb3/pkg/controller/appdefinition/deploy_test.go#L51

But because how we call the handler code in production and test different, the test won't be catch it. It is a lesson to learn.